src/signals-unix.c

```diff
@@ -783,6 +783,17 @@ static void *signal_listener(void *arg)
                     }
                     jl_set_safe_restore(old_buf);
 
+                    jl_ptls_t ptls = jl_all_tls_states[i];
+
+                    // store threadid but add 1 as 0 is preserved to indicate end of block
+                    bt_data_prof[bt_size_cur++].uintptr = ptls->tid + 1;
+
+                    // store task id
+                    bt_data_prof[bt_size_cur++].uintptr = ptls->current_task;
+
+                    // store time
+                    bt_data_prof[bt_size_cur++].uintptr = jl_hrtime();
```

Comment:
@vtjnash is this what you had in mind? I wasn't sure whether this was fast enough to not slow down the profiler significantly?

Reply:
What's the use for the timestamp?

Reply:
It was a suggestion @vtjnash made in passing. There's no use for it in this PR but my guess is it would be useful for looking at profile jitter, and now order of thread sampling given it's shuffled. But it does use up an IP and slow the profiler a bit, so I'm not sure

Reply:
The question is whether to use the OS level high resolution time, or the CPU cycle clock.

The latter might be cheaper, but that is hard to tell not knowing what the goal of the timestamp is.

Reply:
I believe the suggestion was a CPU cycle clock, but I tried to find examples to copy in base and couldn't find one

Reply:
Well that's solvable https://github.com/JuliaLang/julia/blob/b46df09eb651eba97776fe73e8044e0456e81320/src/julia_internal.h#L124

Reply:
Don't know how that evaded my ctrl-f's before. Thanks